### PR TITLE
Fix links to Kubernetes API conventions doc.

### PR DESCRIPTION
**Pull request by shields-fn**

### CODE REVIEW

**Comment by pdecat** [comment]
The links should probably target tags or commit hashes to avoid this sort of issue in the future.

@alexsomesan WDYT?

**Comment by alexsomesan** [comment]
Yes, I think that's a more robust approach.
You can find out the commit ID of the currently vendored k8s.packages in the `go.mod` file.

File: vendor/k8s.io/api/apps/v1/types_swagger_doc_generated.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ package v1
 // AUTO-GENERATED FUNCTIONS START HERE. DO NOT EDIT.
 var map_ControllerRevision = map[string]string{
 	"":         "ControllerRevision implements an immutable snapshot of state data. Clients are responsible for serializing and deserializing the objects that contain their internal state. Once a ControllerRevision has been successfully created, it can not be updated. The API Server will fail validation of all requests that attempt to mutate the Data field. ControllerRevisions may, however, be deleted. Note that, due to its use by both the DaemonSet and StatefulSet controllers for update and rollback, this object is beta. However, it may be subject to name and representation changes in future releases, and clients should not depend on its stability. It is primarily for internal use by controllers.",
-	"metadata": "Standard object's metadata. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata",
+	"metadata": "Standard object's metadata. More info: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#metadata",
```
[the remark this answers is not in the document]
Comment (shields-fn) [reply vendor/k8s.io/api/apps/v1/types_swagger_doc_generated.go]: Reverted.

**Comment by shields-fn** [comment]
Using a fixed commit id would be robust against changing paths in the future, but it means we will be linking to old versions of the documentation. Wouldn't you rather have either a working link or an obviously broken link that gets fixed, rather than a subtly out-of-date link?

**Pull request closed without merging**